File: php/src/Infrastructure/Controller/GreetingController.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Infrastructure\Controller;

use Alxarafe\App\Application\UseCase\CreateGreetingUseCase;
use Alxarafe\App\Application\UseCase\GetGreetingUseCase;
use flight\Engine;

final class GreetingController
{
    public function __construct(
        private CreateGreetingUseCase $createUseCase,
        private GetGreetingUseCase $getUseCase,
    ) {
    }

    /** @phpstan-ignore-next-line */
    public function greet(Engine $app): void
    {
        $name = $app->request()->query->name ?? 'World';

        $greeting = $this->createUseCase->execute($name);

        $app->json([
            'id' => $greeting->getId(),
            'message' => $greeting->getMessage(),
            'createdAt' => $greeting->getCreatedAt()->format('c'),
        ]);
    }

    /** @phpstan-ignore-next-line */
    public function getAll(Engine $app): void
    {
        $greetings = $this->getUseCase->findAll();

        $data = array_map(fn ($g) => [
            'id' => $g->getId(),
            'message' => $g->getMessage(),
            'createdAt' => $g->getCreatedAt()->format('c'),
        ], $greetings);

        $app->json($data);
    }
}

File: java/src/main/java/com/alxarafe/app/infrastructure/controller/GreetingController.java
package com.alxarafe.app.infrastructure.controller;

import com.alxarafe.app.application.usecase.CreateGreetingUseCase;
import com.alxarafe.app.application.usecase.GetGreetingUseCase;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.time.Instant;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api")
public class GreetingController {

    private final CreateGreetingUseCase createUseCase;
    private final GetGreetingUseCase getUseCase;

    public GreetingController(CreateGreetingUseCase createUseCase, GetGreetingUseCase getUseCase) {
        this.createUseCase = createUseCase;
        this.getUseCase = getUseCase;
    }

    @GetMapping("/health")
    public ResponseEntity<Map<String, Object>> health() {
        Map<String, Object> body = new LinkedHashMap<>();
        body.put("status", "ok");
        body.put("timestamp", Instant.now().toString());
        return ResponseEntity.ok(body);
    }

    @GetMapping("/greet")
    public ResponseEntity<Map<String, Object>> greet(@RequestParam(defaultValue = "World") String name) {
        var greeting = createUseCase.execute(name);
        Map<String, Object> body = new LinkedHashMap<>();
        body.put("id", greeting.getId());
        body.put("message", greeting.getMessage());
        body.put("createdAt", greeting.getCreatedAt().toString());
        return ResponseEntity.ok(body);
    }

    @GetMapping("/greetings")
    public ResponseEntity<List<Map<String, Object>>> greetings() {
        var greetings = getUseCase.findAll();
        var data = greetings.stream()
                .map(g -> {
                    Map<String, Object> item = new LinkedHashMap<>();
                    item.put("id", g.getId());
                    item.put("message", g.getMessage());
                    item.put("createdAt", g.getCreatedAt().toString());
                    return item;
                })
                .toList();
        return ResponseEntity.ok(data);
    }
}
